src/lxc/lxccontainer.c

```diff
@@ -2224,12 +2224,15 @@ static inline bool enter_net_ns(struct lxc_container *c)
 	if (pid < 0)
 		return false;
 
+	if (!switch_to_ns(pid, "net"))
```

Comment:
I think we should be a bit more careful here, as it's not a random choice that we do switch to network namespace after user namespace here. For example, you can do `unshare -U -n` as an unprivileged user, but you can not do `unshare -n`. Strictly speaking, you can change user namespace if your current user has `CAP_SYS_ADMIN` in the target user namespace, while for network namespace you need to have `CAP_SYS_ADMIN` in current user namespace **and** also user namespace that owns target network namespace.

I'm pretty sure that with this change we should get a degradation for a case when you run unprivileged container from an unprivileged user and try to list IP address/network interfaces. (Please, correct me if I'm wrong.)

Reply:
Yes I see that there may be a degradation when starting as an unprivileged user. I see that nsenter enters the namespaces in two passes for this reason (https://github.com/util-linux/util-linux/commit/854d0fef7f2bbae6e13a2b2ec8116b7e29541d45), I can modify the code to do something similar.